decidim-core/vendor/assets/javascripts/form_datepicker.js.es6

```diff
@@ -20,7 +20,7 @@
         leftArrow: '<<',
         rightArrow: '>>'
       }).on('changeDate', (ev) => {
-        $(ev.target).siblings('input').val(exports.moment.utc(ev.date));
+        $(ev.target).siblings('input').val(exports.moment(ev.date));
```

Comment:
Yes, Decidim works with any time zone! (I think the default is UTC). By "correct" I mean the timezone where the Decidim app is supposed to be (in Barcelona that would be CET/CEST).

I didn't thought about a situation where the app is in one timezone and the user in a different one (your case). I guess the right think to do would be get the timezone from the user and use that with JavaScript?